source/adapters/level_zero/command_buffer.cpp

```diff
@@ -489,11 +597,12 @@ void ur_exp_command_buffer_handle_t_::registerSyncPoint(
 
 ze_command_list_handle_t
 ur_exp_command_buffer_handle_t_::chooseCommandList(bool PreferCopyEngine) {
-  if (PreferCopyEngine && this->useCopyEngine() && !this->IsInOrderCmdList) {
+  if (PreferCopyEngine && this->useCopyEngine()) {
     // We indicate that ZeCopyCommandList contains commands to be submitted.
     this->MCopyCommandListEmpty = false;
     return this->ZeCopyCommandList;
   }
+  this->MComputeCommandListEmpty = false;
   return this->ZeComputeCommandList;
```

Comment:
A bit out of scope, but while we're touching this function can we get rid the `this->` pointer dereferences, they don't look necessary.